pkg/reconciler/v1alpha1/serverlessservice/serverlessservice.go

```diff
@@ -199,39 +199,38 @@ func (r *reconciler) reconcilePublicService(ctx context.Context, sks *netv1alpha
 	sn := names.PublicService(sks.Name)
 	srv, err := r.serviceLister.Services(sks.Namespace).Get(sn)
 	if errors.IsNotFound(err) {
-		logger.Infof("K8s service %q does not exist; creating.", sn)
+		logger.Infof("K8s service %s does not exist; creating.", sn)
 		// We've just created the service, so it has no endpoints.
 		sks.Status.MarkEndpointsNotReady("CreatingPublicService")
 		srv = resources.MakePublicService(sks)
 		_, err := r.KubeClientSet.CoreV1().Services(sks.Namespace).Create(srv)
 		if err != nil {
-			logger.Errorw(fmt.Sprintf("Error creating K8s Service %s: ", sn), zap.Error(err))
+			logger.Errorw(fmt.Sprint("Error creating K8s Service:", sn), zap.Error(err))
 			return err
 		}
-		logger.Infof("Created K8s service: %q", sn)
+		logger.Info("Created K8s service:", sn)
 	} else if err != nil {
-		logger.Errorw(fmt.Sprintf("Error getting K8s Service %s: ", sn), zap.Error(err))
+		logger.Errorw(fmt.Sprint("Error getting K8s Service:", sn), zap.Error(err))
 		return err
 	} else if !metav1.IsControlledBy(srv, sks) {
 		sks.Status.MarkEndpointsNotOwned("Service", sn)
-		return fmt.Errorf("SKS: %q does not own Service: %q", sks.Name, sn)
-	} else {
-		tmpl := resources.MakePublicService(sks)
-		want := srv.DeepCopy()
-		want.Spec.Ports = tmpl.Spec.Ports
-		want.Spec.Selector = nil
-
-		sks.Status.MarkEndpointsNotReady("UpdatingPublicService")
-		if !equality.Semantic.DeepEqual(want.Spec, srv.Spec) {
-			logger.Infof("Public K8s Service changed; reconciling: %s", sn)
-			if _, err = r.KubeClientSet.CoreV1().Services(sks.Namespace).Update(want); err != nil {
-				logger.Errorw(fmt.Sprintf("Error updating public K8s Service %s: ", sn), zap.Error(err))
-				return err
-			}
+		return fmt.Errorf("SKS: %s does not own Service: %s", sks.Name, sn)
+	}
+	tmpl := resources.MakePublicService(sks)
+	want := srv.DeepCopy()
+	want.Spec.Ports = tmpl.Spec.Ports
+	want.Spec.Selector = nil
+
+	sks.Status.MarkEndpointsNotReady("UpdatingPublicService")
```

Comment:
I don't think we want this unconditionally.

Reply:
ok, removed.